### PR TITLE
Various Changes

### PR DESCRIPTION
**Pull request by simlu**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -61,31 +64,28 @@ const find = (haystack, searches, pathIn, parents, ctx) => {
...
-} = {}) => {
+module.exports = (needles, opts) => {
+  const ctx = Object.assign({
+    filterFn: undefined,
+    breakFn: undefined,
+    joined: true,
+    escapePaths: true,
+    useArraySelector: true
+  }, opts);
+  assert(Object.keys(ctx).length === 5, 'Unexpected Option provided!');
```
Comment (MacMcIrish) [inline src/index.js]: I understand _why_ you're making this check, it's just unfortunate that we then lose code insight. Ah well, deficiency of the language I suppose

Comment (simlu) [reply src/index.js]: Yeah it's stupid. But I much rather have this check than code insight

File: test/integration.spec.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                ? undefined
+                : logFn('breakFn', log, options.log || ['isMatch', 'matchedBy', 'traversedBy']),
+              filterFn: options.log === null
+                ? undefined
+                : logFn('filterFn', log, options.log || ['isMatch', 'matchedBy', 'traversedBy'])
+            });
+            const result = objectScan(fileContent.needles, opts)(dirInput);
+            // eslint-disable-next-line istanbul-prevent-ignore
+            /* istanbul ignore if */
+            if (fileContent.result === undefined) {
```
Comment (MacMcIrish) [inline test/integration.spec.js]: Was this just used to record the tests? No longer necessary?

File: test/integration.spec.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            const options = fileContent.options || {};
+            const log = options.log === null ? null : [];
+            const opts = Object.assign({}, options.args, {
+              breakFn: options.log === null
+                ? undefined
+                : logFn('breakFn', log, options.log || ['isMatch', 'matchedBy', 'traversedBy']),
+              filterFn: options.log === null
+                ? undefined
+                : logFn('filterFn', log, options.log || ['isMatch', 'matchedBy', 'traversedBy'])
+            });
```
Comment (MacMcIrish) [inline test/integration.spec.js]: Thoughts on removing the redundancy?
```js
['breakFn', 'filterFn'].reduce((p, c) => Object.assign(p, {
  [c]: options.log === null
    ? undefined
    : logFn(c, log, options.log || ['isMatch', 'matchedBy', 'traversedBy']
})), {})
```

File: test/integration.spec.js (hunk 2 above)
Comment (simlu) [reply test/integration.spec.js]: It's really handy to have when recording tests. Especially with the formatting

File: test/integration.spec.js (hunk 3 above)
Comment (simlu) [reply test/integration.spec.js]: done

File: test/integration.spec.js (hunk 2 above)
Comment (simlu) [reply test/integration.spec.js]: added a comment

**Pull request merged**